src/H5Gdeprec.c

```diff
@@ -197,7 +196,7 @@ H5Gcreate1(hid_t loc_id, const char *name, size_t size_hint)
             HGOTO_ERROR(H5E_PLIST, H5E_CANTGET, H5I_INVALID_HID, "can't get group info")
 
         /* Set the non-default local heap size hint */
-        H5_CHECKED_ASSIGN(ginfo.lheap_size_hint, uint32_t, size_hint, size_t);
+        ginfo.lheap_size_hint = size_hint;
```

Comment:
Why the change here?   size_t is probably larger than uint32_t and would be truncated if too large.

Reply:
I switched this because I updated the struct field to use size_t, but that's being switched back.